components/script/dom/element.rs

```diff
@@ -2974,3 +2991,53 @@ pub fn cors_setting_for_element(element: &Element) -> Option<CorsSettings> {
         }
     })
 }
+
+pub fn lang_check(css_lang: &Box<str>, element_lang: String) -> bool {
+    let css_lang_tags: Vec<&str> = css_lang.split('\x2d').collect();
+    let element_lang_tags: Vec<&str> = element_lang.split('\x2d').collect();
+
+    if !(css_lang_tags[0].eq_ignore_ascii_case(element_lang_tags[0])) {
+        return false;
+    }
+
+    let mut css_tag_index = 1;
+    let mut tag_index = 1;
+
+    let lang_tag_length = element_lang_tags.len();
+    let css_tag_size = css_lang_tags.len();
+    let is_match;
+
+    loop {
+        if lang_tag_length <= tag_index {
+            is_match = false;
+            break;
+        }
+
+        if css_tag_size <= css_tag_index {
+            is_match = true;
+            break;
+        }
+
+        let css_tag = css_lang_tags[css_tag_index];
+        let element_tag = element_lang_tags[tag_index];
+
+        if css_tag == "*" {
+            css_tag_index += 1;
+            continue;
+        }
+
+        if element_tag.len() == 1 {
+            is_match = false;
+            break;
+        }
+
+        if element_tag.eq_ignore_ascii_case(css_tag) {
+            css_tag_index += 1;
+            tag_index += 1;
+        } else {
+            is_match = false;
+            break;
+        }
+    }
+    return is_match;
```

Comment:
No need for an explicit `return` keyword here; Rust automatically returns the last expression in a function.